File: lab/python/numeros_perfeitos/numeros_perfeitos.py
def eh_numero_perfeito():
    n = int(input("DIGITE O VALOR DE N: "))
    soma = 0

    for i in range(1, n):
        if n % i == 0:
            soma += i

    if soma == n:
        return True
    return False


print(eh_numero_perfeito())

File: lab/c/numeros_perfeitos/numeros_perfeitos.c
#include <stdio.h>

int eh_numero_perfeito();

int main(){ 
    printf("%d\n", eh_numero_perfeito());
    return 0;
}


int eh_numero_perfeito(){
    int soma = 0;
    int n;

    printf("DIGITE O VALOR DE N: ");
    scanf("%d", &n);

    for (int i = 1; i < n; i++){
        if (n % i == 0){
            soma += i;
        }
    }

    if (soma == n) {
        return 1;
    }

    return 0;
}
